iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/IoTDBShutdownHook.java

```diff
@@ -120,7 +124,7 @@ public void run() {
     try (ConfigNodeClient client =
         ConfigNodeClientManager.getInstance().borrowClient(ConfigNodeInfo.CONFIG_REGION_ID)) {
       isReportSuccess =
-          client.reportDataNodeShutdown(DataNode.generateDataNodeLocation()).getCode()
+          client.reportDataNodeShutdown(nodeLocation).getCode()
```

Comment:
What's the difference between "call DataNode.generateDataNodeLocation() here" and "pass the nodeLocation in when constructing the IoTDBShutdownHook" ?

Reply:
Less singleton pattern?

Reply:
It's fine, just ask for curious.